Frends.AzureBlobStorage.UploadBlob/Frends.AzureBlobStorage.UploadBlob.Tests/UnitTestsBlock.cs

```diff
@@ -0,0 +1,637 @@
+            ContainerName = _containerName,
+            BlobType = AzureBlobType.Block,
+            ConnectionString = _connectionString,
+            CreateContainerIfItDoesNotExist = true,
+            FileEncoding = "utf-8",
+            HandleExistingFile = HandleExistingFile.Overwrite,
+            TenantID = default,
+            ApplicationID = default,
+            StorageAccountName = default,
+            ClientSecret = default,
+            ContentType = default,
+            PageMaxSize = default,
+            PageOffset = default,
+            ParallelOperations = default,
+        };
+
+        var container = GetBlobContainer(_connectionString, _containerName);
+
+        foreach (var sp in searchPatterns)
+        {
+            var input = new Source
+            {
+                SourceType = UploadSourceType.Directory,
+                SourceDirectory = _testFileDir,
+                SearchPattern = sp,
+                ContentsOnly = default,
+                Tags = default,
+                Compress = default,
+                SourceFile = default,
+                BlobName = default,
+                BlobFolderName = default,
+            };
+
+            // connection string
+            var resultWithTags = await AzureBlobStorage.UploadBlob(input, destinationCS, options, default);
+            Assert.IsTrue(resultWithTags.Success);
+            Assert.IsTrue(resultWithTags.Data.ContainsValue($"{container.Uri}/TestFiles/SearchPattern.xml"));
+            Assert.IsTrue(await container.GetBlobClient("TestFiles/SearchPattern.xml").ExistsAsync(), "Uploaded testfile.txt blob should exist");
+            Assert.IsTrue(resultWithTags.Data.ContainsValue($"{container.Uri}/TestFiles/SearchPattern.xml"));
+            Assert.IsTrue(await container.GetBlobClient("TestFiles/SearchPattern.xml").ExistsAsync(), "Uploaded SearchPattern.xml blob should exist");
+        }
+    }
+
+    private static BlobContainerClient GetBlobContainer(string connectionString, string containerName)
+    {
+        var blobServiceClient = new BlobServiceClient(connectionString);
+        return blobServiceClient.GetBlobContainerClient(containerName);
+    }
+
+    private void CreateFiles()
+    {
+        Directory.CreateDirectory(_testFileDir);
+
+        // 512 byte file
+        File.WriteAllText(_testfile, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque non sem quis orci rutrum hendrerit. Fusce ultricies cursus ante nec bibendum. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam libero massa, viverra id suscipit in, tincidunt sit amet urna. Vestibulum gravida a massa eu molestie. Phasellus volutpat neque vitae enim molestie, vitae pharetra massa varius. Phasellus ante nulla, faucibus nec tristique eu, dignissim quis magna. Sed vitae sodales ipsum. Ut et maximus nibh. Etiam dui.");
+        File.WriteAllText(_testfile2, "More texts.");
+        File.WriteAllText(_testfile3, "Even more texts.");
+    }
```

Comment:
Please create Helper class for help functions. No need to have this method in every Test class.